Scripts/extras/archive_downloader.py

```diff
@@ -0,0 +1,167 @@
+        return tgt_filename
+
+    print(f"{INDENT}Downloading {src_url}")
+    if os.path.isfile(tgt_filename):
+        os.remove(tgt_filename)
+    urllib.request.urlretrieve(src_url, tgt_filename)
+
+    # Calculate the downloaded file hash 
+    downloaded_hash = hash_file(file_path=tgt_filename,
+                                hash_algorithm=src_zip_hash_algorithm)
+    if src_zip_hash and src_zip_hash != downloaded_hash:
+        raise RuntimeError(f"Hash {src_zip_hash_algorithm} verification failed for {tgt_filename}")
+
+    print(f"{INDENT}Package hash : ({src_zip_hash_algorithm}) {downloaded_hash}")
+
+    return tgt_filename
+
+def extract_package(src_package_file: str, target_folder:str):
+
+    src_package_file_abs = os.path.realpath(src_package_file)
+    target_folder_abs = os.path.realpath(target_folder)
+
+    if not os.path.isfile(src_package_file_abs):
+        raise FileNotFoundError(f"Package to extract '{src_package_file_abs}'' does not exist.")
+    
+    base_filename = os.path.basename(src_package_file_abs)
+
+    package_name, package_ext = os.path.splitext(src_package_file_abs)
+
+    destination_path = os.path.join(target_folder_abs, package_name)
+
+    print(f"{INDENT}Extracting {src_package_file_abs} to {destination_path}")
+
+    if package_ext in ARCHIVE_EXTS_ZIP:
+        import zipfile
+        with zipfile.ZipFile(src_package_file_abs, 'r') as dep_zip:
+            dep_zip.extractall(destination_path)
+    elif package_ext in ARCHIVE_EXTS_TAR:
+        import tarfile
+        with tarfile.open(src_package_file_abs) as tar_file:
+            tarfile.extractall(destination_path)
+    elif package_ext in ARCHIVE_EXTS_7ZIP:
+        try:
+            os.makedirs(destination_path, exist_ok=True)
+            subprocess.call(f"7z x -y {src_package_file_abs}", cwd=destination_path)
+        except Exception:
+            raise RuntimeError(f"Archive file {src_package_file_abs} requires 7Zip to be installed and on the command path. ")
+            
+    else:
+        raise RuntimeError(f"Unsupported package extension: {package_ext}")
+
+    return destination_path
+
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Download, verify hash, and unpack remote zip file")
+    parser.add_argument('src_url', 
+                        help='The download url of the zip package to download',
+                        nargs=1)
+    parser.add_argument('--hash-algorithm',
+                        help='The hash algorithm to use to calculate the fingerprint',
+                        default='sha256',
+                        required=False)
+    parser.add_argument('--hash',
+                        help='The hash fingerprint to validate against',
+                        default='',
+                        required=False)
+
+    parser.add_argument('--target-folder',
+                        help='The target location for the download',
+                        required=True)
```

Comment:
For consistency with the other options, can you make the positional argument use a dash("src-url") instead of an underscore("src_url")